### ISSUE TITLE
Python process frozen after cancel select a script

### ISSUE THREAD

**Issue by gainskills** [brentvollebregt/auto-py-to-exe#56]
Python 3.7.3
OS: MAC 10.14.5
version of auto-py-to-exe and its dependencies: Eel-0.10.4 altgraph-0.16.1 auto-py-to-exe-2.5.3 bottle-0.12.16 bottle-websocket-0.2.9 future-0.17.1 gevent-1.4.0 gevent-websocket-0.10.1 greenlet-0.4.15 macholib-1.11 pefile-2019.4.18 pyinstaller-3.4 whichcraft-0.5.

Step:
1. Run cmd:
autopytoexe script.py 

2. Try to select a script with 'Script location'
3. In the file browser window, it failed to locate .py file and the program frozen after I click the 'Cancel' button of the browser window (From activity manager, the status of Python process is 'No response').  

Issue is replicable on my laptop.

Thanks,

**Comment by brentvollebregt** [comment]
In step 3, what do you mean by it failing to locate your .py file? It should be able to see all files it has permission to see; you can also change what file types are being searched for (in the selection dialog).

I tried these steps on Windows 10 and everything that occurred was expected; a blank script location was inserted after canceling (no freeze).

I don't have access to a Mac to test this but I have seen something like this occur before in my own testing and it had something to do with privileges and threads on Mac. If you give the script elevated privileges, does this still occur?

**Comment by gainskills** [comment]
@brentvollebregt , thanks for your follow up.
'failling to locate the .py file' means in the selection window, I can't select a .py file, that's why I have to click 'Cancel' button and report this issue.
Could you please let me know if any log can be collect for analysis?
Thanks,

**Comment by brentvollebregt** [comment]
If it is occurring in the file selection then this has something to do with tkinter. If you open up IDLE (or create a new file) and execute:

```python
from tkinter.filedialog import askopenfilename
file_path = askopenfilename(filetypes=[('Python files', '*.py;*.pyw'), ('All files', '*')])
print(file_path)
```

Try to select a python file using the dialog that appears, what occurs?

**Comment by gainskills** [comment]
yes, I agree that the issue is caused by tkinter
here is the screenshot wuth the code:
![image](https://user-images.githubusercontent.com/1937472/58736616-998f4980-8452-11e9-88b1-94e147cc5997.png)
the py file is unable to be selected

**Comment by gainskills** [comment]
I would say https://stackoverflow.com/a/19284439/2701959 figure the issue out, about 

```
file_path = askopenfilename(filetypes=[('Python files', '*.py;*.pyw'), ('All files', '*')])
print(file_path)
```
should be:
`file_path = askopenfilename()`

I tested it and it works:
![image](https://user-images.githubusercontent.com/1937472/58737416-9053ac00-8455-11e9-979e-984d79b87a6a.png)

**Comment by brentvollebregt** [comment]
Oh right, this is a MacOS issue; I didn't realise there is no file type drop down in MacOS. I'll add some checks to leave out filetypes on MacOS.

**Comment by brentvollebregt** [comment]
Cool, i've just made a fix and put it on the master branch. If you try this out and it works, I can deploy to PyPI with a new version.

Thanks heaps by the way, I am never able to test on MacOS but I know a lot of this is supported there.

**Comment by gainskills** [comment]
Thanks for your update.
I created a pull request for the file type and frozen issue, could you please review it?
Thanks again,

**Comment by brentvollebregt** [comment]
Issue resolved in #57. I'll deploy to PyPI later when I get a chance.
